hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/TestContainerReconciliationWithMockDatanodes.java

```diff
@@ -350,13 +350,18 @@ public ContainerProtos.GetContainerChecksumInfoResponseProto getChecksumInfo(lon
      */
     public long checkAndGetDataChecksum(long containerID) {
       KeyValueContainer container = getContainer(containerID);
+      KeyValueContainerData containerData = container.getContainerData();
       long dataChecksum = 0;
       try {
         Optional<ContainerProtos.ContainerChecksumInfo> containerChecksumInfo =
-            handler.getChecksumManager().read(container.getContainerData());
+            handler.getChecksumManager().read(containerData);
         assertTrue(containerChecksumInfo.isPresent());
         dataChecksum = containerChecksumInfo.get().getContainerMerkleTree().getDataChecksum();
-        assertEquals(container.getContainerData().getDataChecksum(), dataChecksum);
+        assertEquals(containerData.getDataChecksum(), dataChecksum);
+        try (DBHandle dbHandle = BlockUtils.getDB(containerData, conf)) {
+          Long dbDataChecksum = dbHandle.getStore().getMetadataTable().get(containerData.getContainerDataChecksumKey());
+          assertEquals(containerData.getDataChecksum(), dbDataChecksum, "DB should have the updated data checksum.");
+        }
```

Comment:
Can we add a helper method to `ContainerMerkleTreeTestUtils` that verifies the checksum matches in memory, tree, and DB? I don't think the parameter list would be too long. Then we can invoke that from other tests as needed.